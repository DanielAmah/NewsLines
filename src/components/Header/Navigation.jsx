import React from 'react';
import { IndexLink, Link, browserHistory } from 'react-router';
import PropTypes from 'prop-types';
/**
 * @description 
 * @class Navigation
 * @extends {React.Component}
 * @constructor
 * @returns {nav} return navbar
 */ 
class Navigation extends React.Component {
  constructor() {
    super();
    this.logout = this.logout.bind(this);
  }
  /**
   * Logs out user from website
   * @method {logout}
   * @param {event} event Takes in onClick event
   * @return {void}
   */
  logout(event) {
    event.preventDefault();
    localStorage.removeItem('User');
    browserHistory.replace('/');
    location.reload();
  }

  render() {
      
    return ( 
      <div>
        <nav className="navbar navbar-inverse navbar-top">
          <div className="container-fluid">
            <div className="navbar-header">
              <button type="button" className="navbar-toggle" data-toggle="collapse" data-target="#myNavbar">
                <span className="icon-bar"></span>
                <span className="icon-bar"></span>
                <span className="icon-bar"></span>                        
              </button>
            <a className="navbar-brand" href="#!">News Link</a>
            </div>
             <div className="collapse navbar-collapse" id="myNavbar">
            <ul className="nav navbar-nav">
              <li>{localStorage.User ? <IndexLink to="/">Sources</IndexLink> : null}</li>
            </ul>
            <ul className="nav navbar-nav navbar-right">
              <li>{localStorage.User ? <Link to="/logout" onClick={this.logout}>Logout</Link> : null} </li>
            </ul>
              </div>
          </div>
        </nav>
      </div>
    );
  }
}

/**
 * Default Props
 */
Navigation.defaultProps = {
  info: {
    imageURL: '',
    name: ''
  }
};
/**
 * Set Props
 */
Navigation.propTypes = {
  info: PropTypes.object,
  imageURL: PropTypes.string,
  name: PropTypes.string
};

export default Navigation;

