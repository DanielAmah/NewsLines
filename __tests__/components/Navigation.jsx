/* global expect jest test*/
import React from 'react';
import { mount } from 'enzyme';
import PropTypes from 'prop-types';
import toJson from 'enzyme-to-json';

import Navigation from '../../src/components/Header/Navigation';
xdescribe('Nav Component', () => {
  test('should match the Nav snapshot', () => {
    const component = mount(<Navigation />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
