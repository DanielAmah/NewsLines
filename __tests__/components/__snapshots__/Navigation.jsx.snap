// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component should match the Nav snapshot 1`] = `
<Navigation
  info={
    Object {
      "imageURL": "",
      "name": "",
    }
  }
>
  <div>
    <nav
      className="navbar navbar-inverse navbar-fixed-top"
    >
      <div
        className="container-fluid"
      >
        <div
          className="navbar-header"
        >
          <a
            className="navbar-brand"
            href="#!"
          >
            News Link
          </a>
        </div>
        <ul
          className="nav navbar-nav"
        >
          <li />
        </ul>
        <ul
          className="nav navbar-nav navbar-right"
        >
          <li>
             
          </li>
        </ul>
      </div>
    </nav>
  </div>
</Navigation>
`;
